src/runtime_src/core/pcie/linux/pcidev.h

```diff
@@ -78,22 +69,18 @@ class pci_device
 
   uint16_t domain =           INVALID_ID;
   uint16_t bus =              INVALID_ID;
-  uint16_t dev =              INVALID_ID;
+  uint16_t dev_no =           INVALID_ID;
   uint16_t func =             INVALID_ID;
-  uint16_t vendor_id =        INVALID_ID;
-  uint16_t device_id =        INVALID_ID;
   uint32_t instance =         INVALID_ID;
   std::string sysfs_name =    ""; // dir name under /sys/bus/pci/devices
   int user_bar =              0;  // BAR mapped in by tools, default is BAR0
   size_t user_bar_size =      0;
-  bool is_mgmt() const
-  {
-      return mgmt;
-  }
+  bool is_mgmt =              false;
   bool is_ready =             false;
 
-  pci_device(const std::string& drv_name, const std::string& sysfs_name);
-  virtual ~pci_device();
+  dev(const drv& driver, const std::string& sysfs_name);
```

Comment:
The compiler will create a default constructor.  So it might be best if you explicitly prevent one from begin created.  Something like:
`dev() = delete;`

OR

If you expect one to be created, then create definition and body that will initializes the class and have ALL other constructors use it.

Reply:
added "dev() = delete" to prevent default constructor from being created by compiler.